txpipe/covariance.py

```diff
@@ -31,23 +33,37 @@ class TXFourierGaussianCovariance(PipelineStage):
     def run(self):
         import pyccl as ccl
         import sacc
+        import tjpcov
+        import threadpoolctl
+
         # read the fiducial cosmology
         cosmo = self.read_cosmology()
 
         # read binning
-        sacc_data = self.read_sacc()
-
-        # read the n(z) and f_sky from the source summary stats
-        nz, n_eff, n_lens, sigma_e, fsky, N_tomo_bins = self.read_number_statistics()
-
-        # calculate the overall total C_ell values, including noise
-        theory_c_ell = self.compute_theory_c_ell(cosmo, nz, sacc_data)
-        noise_c_ell = self.compute_noise_c_ell(n_eff, sigma_e, n_lens, sacc_data)
-
-        # compute covariance
-        cov = self.compute_covariance(sacc_data, theory_c_ell, noise_c_ell, fsky)
-        self.save_outputs(sacc_data, cov)
-        #self.save_outputs_firecrown(C,data_vector,ells,nz,binning_info)
+        two_point_data = self.read_sacc()
+
+        # read the n(z) and f_sky from the source summary stats        
+        meta = self.read_number_statistics()
+        
+        # Binning choices. The ell binning is a linear piece with all the
+        # integer values up to 500 (is this level of accuracy needed?)
```

Comment:
can you edit to mention that these choices cam from firecrown, and might be tweaked later?

Reply:
yep